src/PeopleSDKAdapter.integration.test.js

```diff
@@ -0,0 +1,79 @@
+import {createIntegrationTestUser, removeIntegrationTestUser, getPersonHydraID} from './testHelper';
+
+import WebexSDKAdapter from './';
+
+describe('People SDK Adapter', () => {
+  let getPerson$, webexSDKAdapter, subscription, user, userID;
+
+  // Since these are integration tests with live data,
+  // increase the async "idle" timeout so jest doesn't error early.
+  jest.setTimeout(30000);
+
+  beforeAll(async () => {
+    user = await createIntegrationTestUser();
+    userID = getPersonHydraID(user.id);
+    webexSDKAdapter = new WebexSDKAdapter(user.sdk);
+    await webexSDKAdapter.connect();
+  });
+
+  afterAll(async () => {
+    try {
+      await removeIntegrationTestUser(user);
+      await webexSDKAdapter.disconnect();
+    } catch (reason) {
+      // eslint-disable-next-line no-console
+      console.warn('Failed to delete test user for Room SDK Adapter integration tests.', reason);
+    }
+  });
+
+  describe('getPerson() returns', () => {
+    beforeEach(async () => {
+      getPerson$ = webexSDKAdapter.peopleAdapter.getPerson(userID);
+    });
+
+    afterEach(async () => {
+      getPerson$ = null;
+      subscription.unsubscribe();
+    });
+
+    test('a person in a proper shape', async (done) => {
+      subscription = getPerson$.subscribe((person) => {
+        expect(person).toMatchObject({
+          ID: userID,
+          emails: [user.emailAddress],
+          displayName: user.displayName,
+        });
+        done();
+      });
+    });
+
+    test('an updated person status after subscribing', async (done) => {
+      let hasUpdated = false;
+
+      subscription = getPerson$.subscribe(async (person) => {
+        if (!hasUpdated) {
```

Comment:
I tried different operators to skip this check. However, with the way that we are combing the listening part with initial fetch section, we must invoke `getPerson$` and subscribe to it to get all the changes. Hopefully this combination won't cause any additional problems. Otherwise we need to separate the logic.

Reply:
so It does not have anything to do with `rxjs`. our logic is blocking us to remove this check, since we fetch and subscribe at the same time.

Reply:
So here's a suggestion I think might work (but i haven't tested it). We can use the [`skip(1)`](https://rxjs-dev.firebaseapp.com/api/operators/skip) operator to skip the initial data emission. We can update the user after subscribing, which should trigger `getPerson` for testing our scenario. Code would look something like:
```js
subscription = getPerson$.pipe(skip(1)).subscribe((person) => {
  expect(person.status).toEqual('ACTIVE');
   done();
});

await user.sdk.internal.presence.setStatus('active', 1500); // Once this resolves, the subscription above will run
```

Reply:
I exactly tried this approach too. But we still need to subscribe to detect the changes. And skip will not allow us to subscribe to any changes to begin with.

Reply:
You're right, we need to include the async code as part of the pipeline. Otherwise we won't know when the promise resolved. Using [`delayWhen](https://rxjs-dev.firebaseapp.com/api/operators/delayWhen) seems to work. It also makes the code more reliable since we're not mixing async code with observables.

```js
    test('an updated person status after subscribing', (done) => {
      subscription = getPerson$
        .pipe(
          delayWhen(() => from(user.sdk.internal.presence.setStatus('active', 1500))),
          skip(1)
        )
        .subscribe((person) => {
          expect(person.status).toEqual('ACTIVE');
          done();
        });
    });